consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -320,4 +321,48 @@ pub mod tests {
         res[..8].copy_from_slice(&hash.to_be_bytes());
         res
     }
+
+    #[test]
+    fn test_getters() {
+        let engine = Cryptarchia::from_genesis([0; 32], config());
+        let parent = engine.genesis();
+
+        // Get branch directly from HashMap
+        let branch1 = engine
+            .branches
+            .get(&parent)
+            .ok_or("At least one branch should be there")
+            .expect("branch1 is not set");
+
+        let branches = engine.branches();
+
+        // Get branch using getter
+        let branch2 = branches
+            .get(&parent)
+            .ok_or("At least one branch should be there")
+            .expect("branch2 is not set");
+
+        assert_eq!(branch1, branch2);
+        assert_eq!(branch1.id(), branch2.id());
+        assert_eq!(branch1.parent(), branch2.parent());
+        assert_eq!(branch1.slot(), branch2.slot());
+        assert_eq!(branch1.length(), branch2.length());
+
+        let slot = Slot::genesis();
+
+        assert_eq!(slot + 10u64, Slot::from(10));
+
+        let strange_engine = Cryptarchia::from_genesis([100; 32], config());
+        let not_a_parent = strange_engine.genesis();
+
+        _ = match branches
+            .get(&not_a_parent)
+            .ok_or(Error::ParentMissing(not_a_parent))
+        {
+            Ok(_) => panic!("Parent should not be related to this branch"),
+            Err(e) => {
+                assert_ne!(e, Error::ParentMissing(parent));
```

Comment:
why this instead of just checking it returns `None`? I suspect there might be some misunderstanding on what the code is supposed to represent

Reply:
I think I see the problem now. I've used "parent" as an Id to act as a parent only in my test. Because my test uses special case where branch.parent == branch.id == genesisId == [0; 32]. When I call Cryptarchia::from_genesis() this is what will happen. One branch will get created which has ID equal to it's parentID Isn't it ?

When I read apply_header(&self, header: Id, parent: Id, slot: Slot) and I saw how the Error::ParentMissing(parent) was returned after searching using parentID I figured return the same error for my case as my branch was not evolved beyond genesis.

If you think I am too creative for simple case, I will return None and remove the parent terminology.

Reply:
Ok, I see. Yes, genesis is a special case where `parent = id`. It's not that it's too creative for a simple case, it's that outside of the context of applying a block to the ledger this is just an id, not a parent of anything. If we're testing `Branches` is isolation we should not carry meaning that come from downstream users

Reply:
Right, and vice versa, those downstream users won't get misled by my personal interpretation. Fixed at https://github.com/logos-co/nomos-node/pull/657/commits/a588f36681adff5f37782fd76e3e5a306e334cc5